sdk/keyvault/azure-security-keyvault-keys/src/test/java/com/azure/security/keyvault/keys/KeyAsyncClientTest.java

```diff
@@ -30,24 +36,31 @@
     protected void beforeTest() {
         beforeTestSetup();
 
-        if (interceptorManager.isPlaybackMode()) {
-            client = clientSetup(pipeline -> new KeyClientBuilder()
-                .vaultUrl(getEndpoint())
-                .pipeline(pipeline)
-                .buildAsyncClient());
-        } else {
-            client = clientSetup(pipeline -> new KeyClientBuilder()
-                .pipeline(pipeline)
-                .vaultUrl(getEndpoint())
-                .buildAsyncClient());
-        }
+        client = clientSetup(pipeline -> new KeyClientBuilder()
+            .pipeline(pipeline)
+            .vaultUrl(getEndpoint())
+            .buildAsyncClient());
+    }
+
+
+    private void getKeyAsyncClient(HttpClient httpClient,
+        KeyServiceVersion serviceVersion) {
+
+        client = clientSetup(pipeline -> new KeyClientBuilder()
+            .vaultUrl(getEndpoint())
+            .pipeline(pipeline)
+            .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient)
```

Comment:
Same as above. Pipeline and http client both should not be set.

Reply:
Will rework here. Thanks!